diff --git a/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp b/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp
index 3f874a7e02..27a46c9490 100644
--- a/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp
+++ b/subprojects/gst-plugins-bad/ext/zxing/gstzxing.cpp
@@ -448,11 +448,7 @@ gst_zxing_transform_frame_ip (GstVideoFilter * vfilter, GstVideoFrame * frame)
   if (result.isValid ()) {
     GST_DEBUG_OBJECT (zxing, "Symbol found. Text: %s Format: %s",
         result.text ().c_str (),
-#if ZXING_VERSION_MAJOR >= 2
         ToString (result.format ()).c_str ());
-#else
-        ToString (result.format ()));
-#endif
   } else {
     goto out;
   }
